lib/route.js

```diff
@@ -168,11 +170,31 @@ function buildRouting (options) {
   }
 
   function hasRoute ({ options }) {
+    return findRoute(options) !== null
+  }
+
+  function findRoute (options) {
     return router.find(
       options.method,
       options.url || '',
       options.constraints
-    ) !== null
+    )
+  }
+
+  function validateRouteParams (options) {
+    const { method, url, constraints } = options
+    const route = findRoute({ method, url: url || '', constraints })
+    if (route === null) {
+      return false
+    }
+
+    const schemaParams = new Set(Object.keys(route.store.schema.params))
+    const params = new Set(Object.keys(route.params))
+    if (schemaParams.size !== params.size) {
+      return false
+    }
+
+    return [...schemaParams].every(param => params.has(param))
   }
```

Comment:
Overall, I'm +1 on exposing `findRoute` but not in sync with having this validation within core. The scope for the feature seems really specific, that's why the suggestion of having it within a community plugin.